### PR TITLE
JIRA issue checker github action

### PR DESCRIPTION
**Pull request by lghiur**
- This GH Actions aims to check for the presence of a reference to a JIRA issue, either in the branch name or PR title.
- It excludes the checking if the changes are being performed over the GH workflows files or within bin folder

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
JIRA issue number is missing in the branch name and the pull request title. Please include the JIRA issue number in either the branch name or the pull request title.

File: bin/jira-issue-check.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  if [ -z "$BRANCH_ISSUE_NUMBER" ] && [ -z "$PR_ISSUE_NUMBER" ]; then
+    echo "JIRA issue number is missing in both the branch name and the pull request title."
+    echo "valid=false" >> $GITHUB_OUTPUT
+
+    # Comment on the pull request with the error message
+    echo "Commenting on the pull request with the error message..."
+    COMMENT_BODY="JIRA issue number is missing in the branch name and the pull request title. Please include the JIRA issue number in either the branch name or the pull request title."
+    PAYLOAD=$(echo '{}' | jq --arg body "$COMMENT_BODY" '.body = $body')
+    URL=$(jq -r .pull_request.comments_url "$GITHUB_EVENT_PATH")
+    echo $TOKEN
```
Comment (titpetric) [inline bin/jira-issue-check.sh:27]: Left debug?

File: bin/jira-issue-check.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    else
+      echo "JIRA issue number $PR_ISSUE_NUMBER is present in the pull request title."
+      echo "jira_issue_number=$PR_ISSUE_NUMBER" >> $GITHUB_OUTPUT
+    fi
+  fi
+
+  echo "valid=true" >> $GITHUB_OUTPUT
+else
+  echo "Changes to GitHub workflow files or the bin folder detected. Skipping JIRA issue number check."
+  echo "::set-output name=valid::true"  # Set output valid to true
```
Comment (titpetric) [inline bin/jira-issue-check.sh:46]: I believe this is deprecated, the rest uses GITHUB_OUTPUT

File: .github/workflows/jira-issue-check.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  check-jira-issue:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Check JIRA Issue Number
+        id: jira-issue-check
+        run: ./bin/jira-issue-check.sh
```
Comment (buger) [inline .github/workflows/jira-issue-check.yml:18]: You can't refer local files in shared workflows like this. GH action limitations. 

But you can download it from here, see how sbom action works:
https://github.com/TykTechnologies/github-actions/blob/main/.github/workflows/sbom.yaml#L94

**Comment by Tyk-ITS** [comment]
JIRA issue number is missing in the branch name and the pull request title. Please include the JIRA issue number in either the branch name or the pull request title.

**Pull request closed without merging**